src/printf.rs

```diff
@@ -307,6 +309,44 @@ mod test {
         assert_eq!(s, r"+101");
     }
 
+    #[test]
+    fn test_sprintf_array() {
+        let values: Vec<Value> = vec!["hello".into(), "world".into()];
+        let s = sprintf("foo %v", &[Value::Array(values)]);
+        assert!(s.is_ok());
+        let s = s.unwrap();
+        assert_eq!(s, r"foo [hello world]");
+
+        let values: Vec<Value> = vec![42.into(), 100.into()];
+        let s = sprintf("foo %v", &[Value::Array(values)]);
+        assert!(s.is_ok());
+        let s = s.unwrap();
+        assert_eq!(s, r"foo [42 100]");
+    }
+
+    #[test]
+    fn test_sprintf_map() {
+        let mut values: HashMap<String, Value> = HashMap::new();
+        values.insert("hello".into(), "world".into());
+        values.insert("number".into(), 42.into());
+        let s = sprintf("foo %v", &[Value::Map(values)]);
+        assert!(s.is_ok());
+        let s = s.unwrap();
+        // The print order is unpredictable, we can't write
+        // a straight comparison
+        assert!(s.contains("hello:world"));
+        assert!(s.contains("number:42"));
+        assert!(s.starts_with("foo map["));
+        assert!(s.ends_with("]"));
```

Comment:
```suggestion
    assert!(s == "foo map[number:42 hello:world]" || s == "foo map[hello:world number:42]");
```
That's simpler and tests the white space between pairs. (Potentially needs formatting)